agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -218,27 +232,41 @@ func GetTaskProtectionHandler(state dockerstate.TaskEngineState, credentialsMana
 
 		// there are no exceptions but there are failures when getting protection in scheduler
 		if len(response.Failures) > 0 {
-			writeJSONResponse(w, handleFailure(*response.Failures[0].Reason), nil, response.Failures[0], nil, nil, getTaskProtectionRequestType)
+			if len(response.Failures) != ExpectedProtectionResponseLength {
+				err := fmt.Errorf("expect at most %v failure in response, get %v", ExpectedProtectionResponseLength, len(response.Failures))
```

Comment:
I think this error message is not useful to the customer so we should not send it back in the HTTP response. It's an error internal to AWS. Should we send back something more generic like `An unexpected error occurred`?

Reply:
Updated.